client/src/modules/auth/auth.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  auth,
+  AuthProvider,
+  FacebookAuthProvider,
+  GoogleAuthProvider,
+  User,
+  UserCredential,
+} from "../../firebase";
+import { login } from "./client";
+import { SessionModel } from "./models";
+
+export const loginEmailPass = async (
+  email: string,
+  password: string
+): Promise<SessionModel> => {
+  const userCredential = await auth().signInWithEmailAndPassword(
+    email,
+    password
+  );
+  return await authorizeWithServer(userCredential);
+};
+
+export const loginWithGoogle = async (
+  emailRedirectUrl?: string
+): Promise<SessionModel> => {
+  return await signInWithPopup(new GoogleAuthProvider(), emailRedirectUrl);
+};
+export const loginWithFacebook = async (
+  emailRedirectUrl?: string
+): Promise<SessionModel> => {
+  return await signInWithPopup(new FacebookAuthProvider(), emailRedirectUrl);
+};
+
+export const registerEmailPass = async (
+  email: string,
+  password: string,
+  emailRedirectUrl?: string
+): Promise<SessionModel> => {
+  const userCredential = await auth().createUserWithEmailAndPassword(
+    email,
+    password
+  );
+  await sendVerificationEmail(userCredential.user, emailRedirectUrl);
+  return await authorizeWithServer(userCredential);
+};
+
+// TODO: verify this
+// export const applyActionCode = (code: string) => auth().applyActionCode(code);
```

Comment:
Verify if `applyActionCode` can still be used from the client-side, without persistence.